### DISCUSSION TITLE
how to use a serverside module in project

### DISCUSSION THREAD

**Discussion by misakamayako** [nuxt/nuxt#17110] [Questions] [answered]
I am creating my blog website, and I need to pull my markdown file from CDN to the local server to use them(because nuxt-content cant use remote sources), but I can't find a way to execute scheduled tasks or check and download files when I get into a specific blog page.  or more simply, is there a way to execute some not related to routing code without a second server?

**Comment by Flanker72** [comment, marked as answer]
You may create an API which will fetch data from your CDN using the given slug/id. And in the blog page you can use `useAsyncData` to get content from your API. Then you may render it with some kind of markdown renderer, that's simple.
As an alternative, you may create server init script which can schedule the downloader task to fill your static content folder with some data. See https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82 to have an example of such server init script.
